### PR TITLE
[Login- "Continue with WordPress.com" flow] - Add `What is WordPress.com?` button.

### LINKED ISSUE #7212: `What is WordPress.com?` link in "Continue with WordPress.com" flow
Part of **Experiment 2: Add more information about the login process** in pe5sF9-6C-p2

Continue with WordPress.com: Not everyone logging into the app would have an understanding of what their WordPress.com account is. 

- We will be adding a new button link to this screen. 
- Clicking on this would open a webview for users to find out more about what is their WordPress.com account and how to use that to log in to the app.

### PR DESCRIPTION
**Pull request by selanthiraiyan**
Closes: #7212

- [ ] The changes done in [Authenticator library](https://github.com/wordpress-mobile/WordPressAuthenticator-iOS) can be reviewed at https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/646

### Description
This PR adds `What is WordPress.com?` button in the Email entry screen of "Continue with WordPress.com" flow. 

As we cannot expect all users who install the app to be familiar with "WordPress.com" we are adding this button to teach users about "WordPress.com".  

This is part of "**Experiment 2: Add more information about the login process**" in pe5sF9-6C-p2

### Testing instructions 
- Install and launch the app.
- Tap "Continue With WordPress.com" button in the login prolougue screen.
- Ensure that you see "What is WordPress.com?" link under the "Continue" button.
- Tap on the "What is WordPress.com?" link and validate that it opens "https://woocommerce.com/document/what-is-a-wordpress-com-account/" webpage.

### Screenshots

| Before  | After |
| ------------- | ------------- |
|  |   |

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/Authentication/AuthenticationManager.swift [hunk 1]
```diff
@@ -31,6 +31,7 @@ class AuthenticationManager: Authentication {
                                                                 wpcomScheme: ApiCredentials.dotcomAuthScheme,
                                                                 wpcomTermsOfServiceURL: WooConstants.URLs.termsOfService.rawValue,
                                                                 wpcomAPIBaseURL: Settings.wordpressApiBaseURL,
+                                                                whatIsWPComURL: WooConstants.URLs.whatIsWPComURL.rawValue,
```
Comment (jaclync) [inline WooCommerce/Classes/Authentication/AuthenticationManager.swift:38]: do we want to run an AB test on this change as an experiment?

File: RELEASE-NOTES.txt [hunk 2]
```diff
@@ -4,6 +4,7 @@
 -----
 - [**] Order Details: All unpaid orders have a Collect Payment button, which shows a payment method selection screen. Choices are Cash, Card, and Payment Link. [https://github.com/woocommerce/woocommerce-ios/pull/7111]
 - [*] Coupons: Removed the redundant animation when reloading the coupon list. [https://github.com/woocommerce/woocommerce-ios/pull/7137]
+- [internal] Login: Added "What is WordPress.com?" link in "Continue With WordPress.com" flow. [https://github.com/woocommerce/woocommerce-ios/pull/7213]
```
Comment (jaclync) [inline RELEASE-NOTES.txt]: I think this feature is user-facing? if we plan to run an AB test on this, we can add a note about it here

**Review by jaclync — COMMENTED** [review]
Reviewed the authenticator earlier, mostly a question about AB test in this PR!

File: RELEASE-NOTES.txt (hunk 2 above)
Comment (selanthiraiyan) [reply RELEASE-NOTES.txt]: Thank you! I made this user-facing.

File: WooCommerce/Classes/Authentication/AuthenticationManager.swift (hunk 1 above)
Comment (selanthiraiyan) [reply WooCommerce/Classes/Authentication/AuthenticationManager.swift:38]: We have decided to release this without A/B testing. Discussed here - p1657185371754549/1657093471.680409-slack-CGPNUU63E

**Review by jaclync — APPROVED** [review]
LGTM! 🚀 🎉

**Pull request merged**